modules/telegram_module.py

```diff
@@ -36,7 +37,7 @@ def get_wrapper(func):
 
                     def wrapper(bot, update, user_data=None, args=[]):
                         ins = x(context=user_data, update=update, bot=bot)
-                        if len(argspec) != len(args):
+                        if len(argspec) > len(args) or len(argspec) < len(args) and not has_vararg:
```

Comment:
A little difficult to read. I would atleast use parentheses to differentiate the two different cases, as a chain of ors and ands gets unreadable. Also maybe pull out some of the variables like `len(args)`  and `len(argspec)`

Reply:
Rewrote it